sklearn/pipeline.py

```diff
@@ -623,8 +623,11 @@ def classes_(self):
 
     def _more_tags(self):
         # check if first estimator expects pairwise input
-        estimator_tags = self.steps[0][1]._get_tags()
-        return {'pairwise': estimator_tags.get('pairwise', False)}
+        try:
+            pairwise = self.steps[0][1]._get_tags().get('pairwise', False)
```

Comment:
I'd prefer just trying `hasattr` because otherwise it's not clear what may raise the AttributeError (could be e.g. `self.steps`).

Also about `.get('pairwise', False)`: I think it's fair to assume that if `_get_tags` exists, then the key will exist.

Reply:
> Also about .get('pairwise', False): I think it's fair to assume that if _get_tags exists, then the key will exist.

This opens the discussion of "Which tags should we expect to exist". This becomes a bigger question when `BaseEstimator` is not expected to be inherited.

Reply:
> Which tags should we expect to exist

If `_get_tags()` exists, then all of them should exist I'd say

Also, I'm uncomfortable with having the tag's default defined twice: here, and in sklearn.base. The default might not change anytime soon, but this is still a pattern that we should avoid

Reply:
> If _get_tags() exists, then all of them should exist I'd say

This feels like something to be tested in `check_estimator` for "api-only".